internal/integratedservices/services/monitoring/operator.go

```diff
@@ -335,24 +336,26 @@ func (op IntegratedServiceOperator) installPrometheusOperator(
 	if op.migrator != nil {
 		release, err := op.helmService.GetDeployment(ctx, cluster.GetID(), prometheusOperatorReleaseName, op.config.Namespace)
 		if err != nil {
-			return err
-		}
-
-		k8sClientFactory := func() (kubernetes.Interface, error) {
-			kubeConfig, err := cluster.GetK8sConfig()
-			if err != nil {
-				return nil, err
+			if !helm.ErrReleaseNotFound(err) {
```

Comment:
When there is only one additional condition, it's nicer to avoid the arrow pattern and merge the condition with the original branch like this:

```
if err != nil && condition(err) {

} else if err != nil { // if necessary

}
```